privateGPT.py

```diff
@@ -31,9 +33,11 @@ def main():
     # Prepare the LLM
     match model_type:
         case "LlamaCpp":
-            llm = LlamaCpp(model_path=model_path, n_ctx=model_n_ctx, callbacks=callbacks, verbose=False)
+            llm = LlamaCpp(model_path=model_path, n_ctx=model_n_ctx, callbacks=callbacks, verbose=False,n_gpu_layers=n_gpu_layers, use_mlock=use_mlock,top_p=0.9, n_batch=1024)
         case "GPT4All":
             llm = GPT4All(model=model_path, n_ctx=model_n_ctx, backend='gptj', callbacks=callbacks, verbose=False)
+        case "OpenAI":
+            llm = OpenAI(model=model_path,openai_api_key=os.environ.get('OPENAI_API_KEY'),streaming=True,callbacks=callbacks, verbose=False)
```

Comment:
Maybe add checks if the key is present in the .env file?

```python
if not os.environ.get('OPENAI_API_KEY'):
                print("Add your OPENAI_API_KEY to .env variable. Get your OpenAI API key from [here](https://platform.openai.com/account/api-keys).\n")
```
Also, you'll have to check if the key is valid by wrapping the q / a section:

```python
try:
       res = qa(query)
       answer, docs = res['result'], [] if args.hide_source else res['source_documents']
       # .....
       # rest of the code...

except AuthenticationError as e:
    print(f"Warning: An exception occurred. Your OPENAI_API_KEY is invalid: {e.error}")
```